test/clojure/core_test/nan.cljc

```diff
@@ -0,0 +1,15 @@
+(ns clojure.core-test.nan
+  (:require [clojure.test :as t]))
+
+(t/deftest common
+  (t/is (thrown? Exception (NaN? nil)))
+  (t/is (thrown? Exception (NaN? "##NaN")))
+  (t/are [in ex] (= ex (NaN? in))
+         0 false
+         1.0 false
+         -1.0 false
+         (Double/valueOf 1.0) false
+         (Double/valueOf 1.0) false
```

Comment:
Can we use `(double 1.0)` instead, since that doesn't rely on JVM interop?

Reply:
Thank you, that's **much** better!